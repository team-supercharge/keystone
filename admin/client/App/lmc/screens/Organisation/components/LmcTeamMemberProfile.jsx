import React from 'react'
import PropTypes from 'prop-types'
import Selectors from '../../../selectors/index'
import { isBrowser, isTablet } from 'react-device-detect'
import { connect } from 'react-redux'
import { Link } from 'react-router'
import { GlyphButton } from '../../../../elemental'

const renderEditButton = (link) => {
    return (isBrowser || isTablet) ? (
        <GlyphButton
            component={Link}
            glyph='pencil'
            position='left'
            style={styles.editButton}
            to={link}
        >
            Edit
        </GlyphButton>
    ) : (
        <GlyphButton
            component={Link}
            glyph='pencil'
            position='default'
            style={styles.editButton}
            to={link}
        />
    )
}

export const LmcTeamMemberProfile = ({ profile, selectedUser }) => {
    if (!profile) return null
    const editLink = `${Keystone.adminPath}/users/${selectedUser}`
    const role = profile.role === 'carer' ? 'Carer' : 'Carehome Admin'

    return (
        <div>
            <div 
                className='lmc-profile-picture__large' 
                style={{ 
                    ...styles.image, 
                    background: `url(${profile.picture || PLACEHOLDER_IMAGE})` 
                }} 
            />
            <div className='lmc-profile-main-info'>
                <div style={styles.basicInfoContainer}>
                    <span style={styles.name}>
                        { `${profile.name.first} ${profile.name.last}` }
                    </span>
                    <span style={styles.basicInfoText}>
                        Role: { role }
                    </span>
                    <span style={styles.basicInfoText} >
                        { profile.email }
                    </span>
                    <span style={{ ...styles.basicInfoText, marginTop: 20 }}>
                        { `${profile.logCount} logs this month` }
                    </span>
                </div>
                { renderEditButton(editLink) }
            </div>
        </div>
    )
}

const PLACEHOLDER_IMAGE = 'https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png'

const styles = {
    basicInfoContainer: {
        width: '100%',
        textAlign: 'center',
        display: 'flex',
        flexDirection: 'column'
    },
    basicInfoText: {
        color: '#999999',
        fontSize: 18,
        fontWeight: 300,
    },
    editButton: {
        float: 'right',
        position: 'absolute',
        top: 20,
        right: 20,
    },
    image: {
        position: 'relative',
        zIndex: 1,
    },
    name: {
        fontWeight: 600,
        fontSize: 24,
    }
}

LmcTeamMemberProfile.propTypes = {
    selectedUser: PropTypes.string.isRequired,
    profile: PropTypes.object
}

const mapStateToProps = (state) => {
    return {
        profile: Selectors.getSelectedUserProfile(state)
    }
}

export default connect(mapStateToProps)(LmcTeamMemberProfile)