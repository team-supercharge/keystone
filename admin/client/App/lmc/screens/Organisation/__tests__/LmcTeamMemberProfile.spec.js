import React from 'react'
import { shallow } from 'enzyme'
import { LmcTeamMemberProfile } from '../components/LmcTeamMemberProfile.jsx'

describe('LmcTeamMemberProfile', () => {
    let wrapper
    let selectedUser
    let selectedUserProfile
    let savedKeystone

    beforeAll(() => {
        savedKeystone = global.Keystone
        global.Keystone = { adminPath: '/admin' }
    })

    beforeEach(() => {
        selectedUser = 'TestId'
        selectedUserProfile = {
            name: { first: 'Test', last: 'TeamMember' },
            id: selectedUser,
            active: true,
            picture: 'TestPictureUrl',
        }
        wrapper = shallow(
            <LmcTeamMemberProfile
                selectedUser={selectedUser}
                selectedUserProfile={selectedUserProfile}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders a link to edit the team member information', () => {
        const link = wrapper.find('Link')
        expect(link.props().children).toEqual('Edit Information')
        expect(link.props().to).toEqual(`${Keystone.adminPath}/users/${selectedUser}`)
    })

    afterAll(() => {
        global.Keystone = savedKeystone
    })
})