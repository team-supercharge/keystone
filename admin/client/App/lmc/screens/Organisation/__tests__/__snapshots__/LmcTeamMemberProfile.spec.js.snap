// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcTeamMemberProfile renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Button
      aphroditeStyles={Array []}
      color="default"
      variant="fill"
    >
      <Link
        onlyActiveOnIndex={false}
        style={
          Object {
            "color": "black",
            "display": "block",
            "height": "100%",
            "textDecoration": "none",
          }
        }
        to="/admin/users/TestId"
      >
        Edit Information
      </Link>
    </Button>
  </div>,
  "nodes": Array [
    <div>
      <Button
        aphroditeStyles={Array []}
        color="default"
        variant="fill"
      >
        <Link
          onlyActiveOnIndex={false}
          style={
            Object {
              "color": "black",
              "display": "block",
              "height": "100%",
              "textDecoration": "none",
            }
          }
          to="/admin/users/TestId"
        >
          Edit Information
        </Link>
      </Button>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcTeamMemberProfile
        selectedUser="TestId"
        selectedUserProfile={
          Object {
            "active": true,
            "id": "TestId",
            "name": Object {
              "first": "Test",
              "last": "TeamMember",
            },
            "picture": "TestPictureUrl",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcTeamMemberProfile {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "selectedUser": "TestId",
          "selectedUserProfile": Object {
            "active": true,
            "id": "TestId",
            "name": Object {
              "first": "Test",
              "last": "TeamMember",
            },
            "picture": "TestPictureUrl",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Button
            aphroditeStyles={Array []}
            color="default"
            variant="fill"
          >
            <Link
              onlyActiveOnIndex={false}
              style={
                Object {
                  "color": "black",
                  "display": "block",
                  "height": "100%",
                  "textDecoration": "none",
                }
              }
              to="/admin/users/TestId"
            >
              Edit Information
            </Link>
          </Button>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Button
            aphroditeStyles={Array []}
            color="default"
            variant="fill"
          >
            <Link
              onlyActiveOnIndex={false}
              style={
                Object {
                  "color": "black",
                  "display": "block",
                  "height": "100%",
                  "textDecoration": "none",
                }
              }
              to="/admin/users/TestId"
            >
              Edit Information
            </Link>
          </Button>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcTeamMemberProfile
    selectedUser="TestId"
    selectedUserProfile={
      Object {
        "active": true,
        "id": "TestId",
        "name": Object {
          "first": "Test",
          "last": "TeamMember",
        },
        "picture": "TestPictureUrl",
      }
    }
  />,
}
`;
